httpx/_content_streams.py

```diff
@@ -95,21 +92,14 @@ def __iter__(self) -> typing.Iterator[bytes]:
     def __aiter__(self) -> typing.AsyncIterator[bytes]:
         raise RuntimeError("Attempted to call a async iterator on an sync stream.")
 
-    def close(self) -> None:
-        if self.close_func is not None:
-            self.close_func()
```

Comment:
Turns out we don't *really* need `close_func` on `IteratorStream`/`AsyncIteratorStream` anymore, which falls out from updating test cases to use `Response(content=<bytes iterator>)`